docs/auth.md

```diff
@@ -0,0 +1,191 @@
+# Serving Certificates, Authentication, and Authorization
+
+This document outlines how the service catalog API server handles
+authentication and authorization, and the steps needed to set this up in
+your cluster.
+
+Additionally, it provides instructions for generating the serving
+certificates used to securely serve the service catalog API.
+
+## Quickstart
+
+1. Ensure that you have a ConfigMap in the `kube-system` namespace named
+   `extensions-apiserver-authentication`
+2. Ensure that there is a `client-ca-file` key in that ConfigMap.  If not,
+   you'll need a client CA file (see [generating
+   certificates](#generating-certificates) below).  This can be passed to
+   the service catalog API server using `--client-ca-file`.
+3. Ensure that there is a `requestheader-client-ca-file` key in that
+   ConfigMap. If not, you'll need a RequestHeader client CA file (see
+   [generating certificates](#generating-certificates) below).  This can
+   be passed to the service catalog API server using
+   `--requestheader-client-ca-file`, or the
+   `apiserver.tls.requestHeaderCA` Helm chart option..
+4. If not running on the cluster, ensure that you pass in a Kubeconfig
+   file to authenticate with the main cluster using the
+   `--authentication-kubeconfig` and `--authorization-kubeconfig`.
+
+## Generating certificates
+
+The Kubernetes documentation has a [detailed
+section](https://kubernetes.io/docs/admin/authentication/#creating-certificates)
+on how to create certificates several different ways.  For convinience,
+we'll reproduce the basics using the `openssl` and `cfssl` commands below
+(you can install `cfssl` using `go get -u
+github.com/cloudflare/cfssl/cmd/...`).
+
+In the common case, you'll want to re-use the CA certificates used by the
+main Kubernetes setup, but you'll need to generate your own serving
+certificates.
+
+In case you need to generate a CA certificate pair yourself, you can do so
+using the following command:
+
+```shell
+$ openssl -x509 -sha256 -new -nodes -days 365 -newkey rsa:2048 -keyout ${PURPOSE}-ca.key -out ${PURPOSE}-ca.crt -subj "/CN=ca"
+$ echo '{"signing":{"default":{"expiry":"43800h","usages":["signing","key encipherment","'${PURPOSE}'"]}}}' > "${PURPOSE}-ca-config.json"
+```
+
+This will generate a certificate and private key for the CA, as well as
+a signing configuration used by `cfssl` below. `$PURPOSE` may be set to
```

Comment:
Move the ${PURPOSE} discussion above the commands because we need it for them to work.

Alternatively, leave it, so when people copy and paste it breaks. But they could stop and file an issue rather than read on.